integrations/zoom/events.go

```diff
@@ -0,0 +1,170 @@
+			zap.String("account_id", accountID),
+			zap.Error(err))
+		common.HTTPError(w, http.StatusInternalServerError)
+		return
+	}
+
+	// Dispatch the event to all connections for potential asynchronous handling.
+	common.DispatchEvent(ctx, h.logger, h.dispatch, akEvent, cids)
+}
+
+// checkRequest validates the Zoom webhook HTTP request by verifying the content type,
+// required headers, and signature and returns the parsed JSON payload if valid.
+// Otherwise it returns nil with an HTTP error response.
+func (h handler) checkRequest(w http.ResponseWriter, r *http.Request) map[string]any {
+	l := h.logger.With(
+		zap.String("url_path", r.URL.Path),
+		zap.String("event_type", r.Header.Get("Zoom-Event")),
+	)
+
+	// Check the request's HTTP headers.
+	ct := r.Header.Get("Content-Type")
+	if !strings.HasPrefix(ct, "application/json") {
+		l.Warn("incoming event: unexpected content type", zap.String("content_type", ct))
+		common.HTTPError(w, http.StatusBadRequest)
+		return nil
+	}
+
+	signature := r.Header.Get("x-zm-signature")
+	timestamp := r.Header.Get("x-zm-request-timestamp")
+
+	if signature == "" || timestamp == "" {
+		l.Warn("incoming event: missing Zoom verification headers")
+		common.HTTPError(w, http.StatusBadRequest)
+		return nil
+	}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		l.Error("incoming event: failed to read request body", zap.Error(err))
+		common.HTTPError(w, http.StatusInternalServerError)
+		return nil
+	}
+	r.Body = io.NopCloser(bytes.NewBuffer(body))
+
+	secret, err := h.signingSecret()
+	if err != nil {
+		l.Error("incoming event: signing secret not found", zap.Error(err))
+		common.HTTPError(w, http.StatusInternalServerError)
+		return nil
+	}
+
+	if secret == "" {
+		// Zoom is not configured, so there's no point
+		// in verifying or accepting the payload.
+		return nil
+	}
+
+	// Verify Zoom signature.
+	if !h.checkSignature(signature, timestamp, secret, body) {
+		l.Warn("incoming event: invalid Zoom signature")
+		common.HTTPError(w, http.StatusUnauthorized)
+		return nil
+	}
+
+	var payload map[string]any
+	if err := json.Unmarshal(body, &payload); err != nil {
+		l.Warn("incoming event: failed to parse JSON body", zap.Error(err))
+		common.HTTPError(w, http.StatusBadRequest)
+		return nil
+	}
+
+	return payload
+}
+
+// retrieves the Zoom secret token from environment variables.
+func (h handler) signingSecret() (string, error) {
+	secret := os.Getenv("ZOOM_SECRET_TOKEN")
+	if secret == "" {
+		return "", errors.New("missing Zoom secret token")
+	}
+	return secret, nil
+}
+
+// checkSignature compares the computed hash of the request body, using a preconfigured
+// Zoom webhook secret token, to the signature provided in the request's `x-zm-signature`
+// header, using HMAC-SHA256 for integrity verification.
+// (based on https://developers.zoom.us/docs/api/webhooks/#verify-with-zooms-header)
+func (h handler) checkSignature(signature, timestamp, secret string, body []byte) bool {
```

Comment:
`handler` isn't used in this function, so it's not needed as a received in this function.

Note that the same can be said about `signingSecret()` above, but that will change as soon as we implement INT-332.